miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditProjectController.java

```diff
@@ -255,6 +255,15 @@ public ModelAndView setupForm(ModelMap model) throws IOException {
     return setupForm(ProjectImpl.UNSAVED_ID, model);
   }
 
+  @RequestMapping(value = "/shortname/{shortName}", method = RequestMethod.GET)
+  public ModelAndView byProjectShortName(@PathVariable String shortName, ModelMap model) throws IOException {
+    Project project = projectService.getProjectByShortName(shortName);
+    if (project == null) {
+      return new ModelAndView("/pages/notFound.jsp", model);
```

Comment:
You could instead annotate a method with `@ExceptionHandler(NotFoundException.class)` and `@ResponseStatus(HttpStatus.NOT_FOUND)` to return the error page, and then throw a `NotFoundException` to trigger it, which would result in the proper HTTP status too.

This method could go in a separate class annotated with `@ControllerAdvice` to apply to all controllers, though you'll want to exclude the REST controllers